tests/real/community.rs

```diff
@@ -795,46 +800,183 @@ async fn test_jobs_with_blocking_handlers() {
+        ),
+    ]);
+    let njobs = job_kind_vs_error_msg.keys().len();
+
+    // clean up
+    let local = "test_panic_and_errors_in_handler";
+    let mut c = Client::connect().await.unwrap();
+    let pattern = format!(r#"*\"args\":\[\"{}\"\]*"#, local);
+    c.requeue(
+        MutationTarget::Retries,
+        MutationFilter::builder().pattern(pattern.as_str()).build(),
+    )
+    .await
+    .unwrap();
+    c.queue_remove(&[local]).await.unwrap();
 
     let mut w = Worker::builder::<io::Error>()
-        .register_blocking_fn("panic_SYNC_handler", |_j| {
-            panic!("Panic inside sync the handler...");
+        //  sync handlers
+        .register_blocking_fn("panic_SYNC_handler_str", |_j| {
+            panic!("panic_SYNC_handler_str");
+        })
+        .register_blocking_fn("panic_SYNC_handler_String", |_j| {
+            panic_any("panic_SYNC_handler_String".to_string());
+        })
+        .register_blocking_fn("panic_SYNC_handler_int", |_j| {
+            panic_any(0);
+        })
+        .register_blocking_fn("error_from_SYNC_handler", |_j| {
+            Err::<(), io::Error>(io::Error::new(
+                io::ErrorKind::Other,
+                "error_from_SYNC_handler",
+            ))
+        })
+        // async handlers
+        .register_fn("panic_ASYNC_handler_str", |_j| async move {
+            panic!("panic_ASYNC_handler_str");
+        })
+        .register_fn("panic_ASYNC_handler_String", |_j| async move {
+            panic_any("panic_ASYNC_handler_String".to_string());
+        })
+        .register_fn("panic_ASYNC_handler_int", |_j| async move {
+            panic_any(0);
         })
-        .register_fn("panic_ASYNC_handler", |_j| async move {
-            panic!("Panic inside async handler...");
+        .register_fn("error_from_ASYNC_handler", |_j| async move {
+            Err::<(), io::Error>(io::Error::new(
+                io::ErrorKind::Other,
+                "error_from_ASYNC_handler",
+            ))
         })
         .connect()
         .await
         .unwrap();
 
-    let mut c = Client::connect().await.unwrap();
-
-    c.enqueue(Job::builder("panic_SYNC_handler").queue(local).build())
-        .await
-        .unwrap();
+    // let's enqueue jobs first time and ...
+    c.enqueue_many(
+        job_kind_vs_error_msg
+            .keys()
+            .map(|&jkind| Job::builder(jkind).queue(local).args([local]).build())
+            .collect::<Vec<_>>(),
+    )
+    .await
+    .unwrap();
+
+    // ... consume all the jobs from the queue and _fail_ them
+    // "in different ways" (see our worker setup above)
+    for _ in 0..njobs {
+        assert!(w.run_one(0, &[local]).await.unwrap());
+    }
 
-    // we _did_ consume and process the job, the processing result itself though
-    // was a failure; however, a panic in the handler was "intercepted" and communicated
-    // to the Faktory server via the FAIL command;
-    // note how the test run is not interrupted with a panic
```

Comment:
Feels worthwhile to preserve this comment in the new code.

Reply:
threw the baby out with the bathwater